pkg/datagatherer/k8s/dynamic.go

```diff
@@ -200,34 +199,22 @@ func (c *ConfigDynamic) newDataGathererWithClient(ctx context.Context, cl dynami
 			informers.WithNamespace(metav1.NamespaceAll),
 			informers.WithTweakListOptions(func(options *metav1.ListOptions) {
 				options.FieldSelector = fieldSelector.String()
-			}))
-		newDataGatherer.nativeSharedInformer = factory
-		informer := informerFunc(factory)
-		informer.AddEventHandler(k8scache.ResourceEventHandlerFuncs{
-			AddFunc: func(obj interface{}) {
-				onAdd(obj, dgCache)
-			},
-			UpdateFunc: func(old, new interface{}) {
-				onUpdate(old, new, dgCache)
-			},
-			DeleteFunc: func(obj interface{}) {
-				onDelete(obj, dgCache)
+			}),
+		)
+		newDataGatherer.informer = informerFunc(factory)
+	} else {
+		factory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(
+			cl,
+			60*time.Second,
+			metav1.NamespaceAll,
+			func(options *metav1.ListOptions) {
+				options.FieldSelector = fieldSelector.String()
 			},
-		})
-		newDataGatherer.informer = informer
-		return newDataGatherer, nil
+		)
+		newDataGatherer.informer = factory.ForResource(c.GroupVersionResource).Informer()
 	}
 
-	factory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(
```

Comment:
We are creating these factories to create just 1 informer.
I still create the factory ( maybe this can be simplified in the future ), but I now let it go out-of-scope and continue with the created informer instead.